modules/core/src/views/view.js

```diff
@@ -64,7 +64,7 @@ export default class View {
     }
 
     // TODO - implement deep equal on view descriptors
-    const viewChanged = deepEqual(this, view);
+    const viewChanged = deepEqual(this.props, view.props);
```

Comment:
Do we still consider this a hack? Should the TODO remain?

Reply:
We are not comparing `id` and `type`, we are assuming they would always be same if props are same? Maybe add a comment...

Hmm, if so why does `deepEqual` on the `props` work when `deepEqual` on `this` fails?

Reply:
`id` and `type` are part of `props`
Because `deepEqual` is not really checking if two objects deep equal... This one only goes one-level deep.